plugins/utils/oil.nix

```diff
@@ -5,224 +5,135 @@
-
-          highlight = highlightsOption;
-
-          defaultFile = helpers.mkNullOrOption types.str ''
-            Fallback icon for files when nvim-web-devicons returns nil.
-          '';
-
-          directory = helpers.mkNullOrOption types.str "Icon for directories.";
-        };
-
-        size = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the size column.
-              The size of the file.
-
-              Adapters: files, ssh
-            '';
-          };
-
-          highlight = highlightsOption;
-        };
-
-        permissions = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the perimissions column.
-              Access permissions of the file.
-
-              Adapters: files, ssh
-              Editable: this column is read/write
-            '';
-          };
-
-          highlight = highlightsOption;
-        };
-
-        ctime = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the ctime column.
-              Change timestamp of the file.
-
-              Adapters: files
-            '';
-          };
-
-          highlight = highlightsOption;
-
-          format = formatOption;
-        };
-
-        mtime = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the mtime column.
-              Last modified time of the file.
-
-              Adapters: files
-            '';
-          };
-
-          highlight = highlightsOption;
-
-          format = formatOption;
-        };
-
-        atime = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the atime column.
-              Last access time of the file.
-
-              Adapters: files
-            '';
-          };
-
-          highlight = highlightsOption;
-
-          format = formatOption;
-        };
-
-        birthtime = {
-          enable = mkOption {
-            type = types.bool;
-            default = false;
-            description = ''
-              Whether to enable the birthtime column.
-              The time the file was created.
-
-              Adapters: files
-            '';
-          };
-
-          highlight = highlightsOption;
+with lib;
+  helpers.neovim-plugin.mkNeovimPlugin config {
+    name = "oil";
+    originalName = "oil.nvim";
+    defaultPackage = pkgs.vimPlugins.oil-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-03-18: remove 2024-05-18
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = [
+      "defaultFileExplorer"
+      "bufOptions"
```

Comment:
Does this work with the suboptions (as bufOptions is _not_ an option currently?)

Reply:
Yes, it is better to explicit each suboption. Now fixed.